packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderContent/ConditionBuilderContent.tsx

```diff
@@ -120,12 +119,17 @@ const ConditionBuilderContent = ({
 
   const onRemove = useCallback(
     (groupId) => {
+      const groups = rootState?.groups?.filter(
```

Comment:
question: is this only removing a single entry from the `groups` array? because if so it would be more efficient to use a combination of `findIndex` and `toSpliced` instead. operations like `filter` don't stop after the first match: it will continue to search the array for additional matches. if we know that `onRemove` only removes one group at a time then i would refactor it to use the other methods. it's a small performance consideration, but it's also just good practice to think about operations in this manner.

if multiple groups are being removed though then i would ignore this comment 😄

Reply:
The result are multiple groups except the removed group.